### ISSUE TITLE
[docs] scaled_dot_product_attention is_causal description is misleading

### ISSUE THREAD

**Issue by MikeTkachuk** [pytorch/pytorch#126873]
### 📚 The doc issue

In scaled_dot_product_attention parameter description, the current wording 
"[is_causal](https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html) (bool) – If true, assumes **upper left causal attention masking** and errors if both attn_mask and is_causal are set."
is misleading. I guess there is a mistake?

### Suggest a potential alternative/fix

It should be lower/left masking (or upper/right depending on the wording)

cc @svekars @brycebortree @jbschlosser @bhosmer @cpuhrsch @erichan1 @drisspg @mikaylagawarecki

**Comment by drisspg** [comment]
Sure we would accept a PR changing this. For context this is to distinguish between the alignment of the lower_triangular mask. We coined the inverse of the terminology used in Flash attention and xformers: https://github.com/facebookresearch/xformers/blob/8b3fc6d2e05af22a38fb79ee5b729d3d8bfaff7b/xformers/attn_bias_utils.py#L95

which uses bottom_right to specify a form of alignment

**Comment by MikeTkachuk** [comment]
is_causal does not involve alignment, so i think 
"[is_causal](https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html) (bool) – If true, assumes **lower triangular causal attention masking** and errors if both attn_mask and is_causal are set." should be safe

**Comment by jbschlosser** [comment]
Alignment comes into play for non-square matrices, I believe. Maybe it makes sense to have a simpler "lower triangular" phrasing for the general case and spell out the alignment explicitly for the non-square case in a follow-up sentence.

**Comment by MikeTkachuk** [comment]
From what I can see, the current implementation simply calls torch.tril(attention) for any shape. Both in the docs pseudocode and in an actual experiment like ` q = torch.ones((2,5)).float(); k = torch.ones((4,5)).float(); torch.nn.functional.scaled_dot_product_attention(q,k,torch.eye(4).float(),is_causal=True)`
I think lower triangular should be enough for both cases
